### PR TITLE
test(text facets): unit and snapshot cases for text facets

### PR DESCRIPTION
**Pull request by deepunbxd005**
n

### CODE REVIEW

File: src/modules/textFacets/__tests__/textFacets.test.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+};
+
+test('Match Snapshot for text facets', () => {
+    const tree = renderer
+        .create(
+            <UnbxdSearchWrapper
+                siteKey="wildearthclone-neto-com-au808941566310465"
+                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
+            >
+                <Products attributesMap={attributesMap} />
```
Comment (vikramgupta1988) [inline src/modules/textFacets/__tests__/textFacets.test.js:52]: Please use textfacet in the snapshot or else it's just a duplicate of product snapshot testing. Also, look why a snapshot is not created completely with products. Try to create a snapshot once the promise is resolved.

File: src/modules/textFacets/__tests__/textFacets.test.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+// establish API mocking before all tests
+beforeAll(() => {
+    window.fetch = jest.fn(() => {
+        return Promise.resolve({
+            json: () => Promise.resolve(searchResponse),
+        });
+    });
+})
+
+const attributesMap = {
```
Comment (sanjithkumar017) [inline src/modules/textFacets/__tests__/textFacets.test.js:37]: Use wildearth's attributeMap. `min_cheapest_default_price` does not exist on wildearth feed.

File: src/modules/textFacets/__tests__/textFacets.test.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        expect(getByText('Scarpa')).toBeInTheDocument();
+        fireEvent.click(getByText("Scarpa"));
+    });
+    await waitFor(() => {
+        expect(getByText("Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots")).toBeInTheDocument();
+        fireEvent.click(getByText("Clear"));
+    });
+});
+
+test('Test text facet click with renderProps', async () => {
```
Comment (sanjithkumar017) [inline src/modules/textFacets/__tests__/textFacets.test.js:119]: How is this render prop way of impl?

File: src/modules/textFacets/__tests__/textFacets.test.js (hunk 2 above)
Comment (vikramgupta1988) [reply src/modules/textFacets/__tests__/textFacets.test.js:37]: @sanjithkumar017  Does it matter when the data and response is mocked?

Comment (vikramgupta1988) [reply src/modules/textFacets/__tests__/textFacets.test.js:37]: @deepunbxd005  Use RRP_Price, we have ti change it everywhere,

File: src/modules/textFacets/__tests__/textFacets.test.js (hunk 1 above)
Comment (deepunbxd005) [reply src/modules/textFacets/__tests__/textFacets.test.js:52]: working on it

File: src/modules/textFacets/__tests__/textFacets.test.js (hunk 3 above)
Comment (Anupama-Unbxd) [reply src/modules/textFacets/__tests__/textFacets.test.js:119]: Please rename the test case if it is not using render props or comment it out if it is not yet implemented completely

Comment (deepunbxd005) [reply src/modules/textFacets/__tests__/textFacets.test.js:119]: yes will update it.

File: src/modules/textFacets/__tests__/textFacets.test.js (hunk 2 above)
Comment (deepunbxd005) [reply src/modules/textFacets/__tests__/textFacets.test.js:37]: done

File: src/modules/textFacets/__tests__/textFacets.test.js (hunk 1 above)
Comment (deepunbxd005) [reply src/modules/textFacets/__tests__/textFacets.test.js:54]: done

**Review by vikramgupta1988 — CHANGES_REQUESTED** [review]

File: src/modules/combinedFacets/__tests__/CombinedFacets.test.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                <div>
+                    <SearchBox defaultSearch="shoes" />
+                </div>
+            </UnbxdSearchWrapper>
+        </>
+    );
+
+    await waitFor(async () => {
+        expect(getByText('$ 200 - $ 300 - 60')).toBeInTheDocument();
+        fireEvent.click(getByText("$ 200 - $ 300 - 60"));
```
[the remark this answers is not in the document]
Comment (deepunbxd005) [reply src/modules/combinedFacets/__tests__/CombinedFacets.test.js:69]: On click of the facet, the appstate changes and the facet gets appended to the url and only then clear button appears, so if Clear button appears then we can be sure that facet got applied and new data is being shown.

File: src/modules/combinedFacets/__tests__/CombinedFacets.test.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    });
+    await waitFor(() => {
+        expect(getByText("Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black")).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (deepunbxd005) [reply src/modules/combinedFacets/__tests__/CombinedFacets.test.js:72]: same

File: src/modules/combinedFacets/__tests__/CombinedFacets.test.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        </>
+    );
+
+    await waitFor(async () => {
+        expect(getByText('Scarpa - 18')).toBeInTheDocument();
+        fireEvent.click(getByText("Scarpa - 18"));
+    });
+    await waitFor(() => {
+        expect(getByText("Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots")).toBeInTheDocument();
+        fireEvent.click(getByText("Clear"));
```
[the remark this answers is not in the document]
Comment (deepunbxd005) [reply src/modules/combinedFacets/__tests__/CombinedFacets.test.js:101]: the appstate goes back to normal state without the filter

File: src/modules/combinedFacets/__tests__/CombinedFacets.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (vikramgupta1988) [reply src/modules/combinedFacets/__tests__/CombinedFacets.test.js:69]: But you are not checking for clear btn in this test case, Right? This test case will pass even if the facets click doesn't work. It should be something like this
- Change the response on the facets click
- Then check for the new products.
- After that click on clear btn to check for the old products.

File: src/modules/combinedFacets/__tests__/CombinedFacets.test.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+            </UnbxdSearchWrapper>
+        </>
+    );
+
+    await waitFor(async () => {
+        expect(getByText('All')).toBeInTheDocument();
+        fireEvent.click(getByText("All"));
+    });
+    await waitFor(() => {
+        expect(getByText("Vasque Breeze All-Terrain GTX Womens Hiking Boots - Gargyle")).toBeInTheDocument();
```
Comment (vikramgupta1988) [inline src/modules/combinedFacets/__tests__/CombinedFacets.test.js:128]: Same here too.

File: src/modules/combinedFacets/__tests__/CombinedFacets.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (deepunbxd005) [reply src/modules/combinedFacets/__tests__/CombinedFacets.test.js:69]: ok

File: src/modules/combinedFacets/__tests__/CombinedFacets.test.js (hunk 7 above)
Comment (deepunbxd005) [reply src/modules/combinedFacets/__tests__/CombinedFacets.test.js:128]: on it

File: src/modules/combinedFacets/__tests__/CombinedFacets.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (deepunbxd005) [reply src/modules/combinedFacets/__tests__/CombinedFacets.test.js:83]: done

File: src/modules/combinedFacets/__tests__/CombinedFacets.test.js (hunk 7 above)
Comment (deepunbxd005) [reply src/modules/combinedFacets/__tests__/CombinedFacets.test.js:142]: done

File: src/modules/textFacets/__tests__/textFacets.test.js (hunk 3 above)
Comment (deepunbxd005) [reply src/modules/textFacets/__tests__/textFacets.test.js]: done

File: src/modules/textFacets/__tests__/textFacets.test.js [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+            </UnbxdSearchWrapper>
+        </>
+    );
+
+    await waitFor(async () => {
+        expect(getByText('Scarpa')).toBeInTheDocument();
+        fireEvent.click(getByText("Scarpa"));
+    });
+    await waitFor(() => {
+        expect(getByText("Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots")).toBeInTheDocument();
```
Comment (vikramgupta1988) [inline src/modules/textFacets/__tests__/textFacets.test.js:122]: Please check the customizations everywhere.

Comment (deepunbxd005) [reply src/modules/textFacets/__tests__/textFacets.test.js:145]: done

**Review by vikramgupta1988 — APPROVED** [review]

**Pull request merged**